ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts

```diff
@@ -20,23 +21,42 @@ export class JobTimingDiagramComponent implements OnInit {
       chartType: 'Timeline',
       dataTable: []
     };
-    this.timelineChart.dataTable.push(['Num', 'Name', 'Start', 'End']);
+    this.timelineChart.dataTable.push([
+      {'type': 'string', 'id': 'Task'},
+      {'type': 'string', 'id': 'Section'},
+      {'type': 'date', 'purpose': 'Start', 'id': 'Start'},
+      {'type': 'date', 'purpose': 'End', 'id': 'End'}
+    ]);
     let counter = 1;
 
     metadata.forEach((task) => {
       if (task.name && task.start) {
-        if (task.end) {
-          this.timelineChart.dataTable.push([counter.toString(), task.name, task.start, task.end]);
+        if (task.executionEvents) {
+          task.executionEvents.forEach((event) => {
+            this.timelineChart.dataTable.push(this.formatRow(task.name, event.name, event));
+          });
         } else {
-          this.timelineChart.dataTable.push([counter.toString(), task.name, task.start, new Date()]);
+          this.timelineChart.dataTable.push(this.formatRow(task.name, task.name, task));
         }
         counter++;
       }
-    })
+    });
+
     this.timelineChart.options = {
-      title: 'Tasks',
-      width: 800,
-      height: counter * 50
+      timeline: {
+        showRowLabels: false,
+        showBarLabels: false
+      },
+      width: this.tabWidth - 65,
+      height: (counter * 42) + 100,
```

Comment:
nit: "counter" is pretty generic - I'd name it something more semantically relevant, such as `numRows`, or whatever it's actually referring to.

Reply:
good point – changed